fabric-network/test/impl/query/queryhandlers.js

```diff
@@ -17,204 +19,154 @@ chai.use(require('chai-as-promised'));
 describe('QueryHandlers', () => {
 	const querySuccessResult = Buffer.from('SUCCESS_QUERY_RESULT');
 	const queryErrorResult = new Error('ERROR_QUERY_RESULT');
-	queryErrorResult.isProposalResponse = true;
 	const queryFailMessage = 'QUERY_FAILED';
-
-	let failedPeers;
-	let errorPeers;
-	let fakeEvaluate;
-	let stubQuery;
-	let stubPeer1;
-	let stubPeer2;
+	const peerInfo = {name: 'peer1', url: 'grpc://fakehost:9999'};
+	const validProposalResponse = {
+		response: {
+			status: 200,
+			payload: querySuccessResult
+		},
+		connection: peerInfo
+	};
+	const invalidProposalResponse = {
+		response: {
+			status: 500,
+			message: queryFailMessage
+		},
+		connection: peerInfo
+	};
+
+	const validProposalResponses = {responses: [validProposalResponse], queryResponses: [querySuccessResult]};
+	const invalidProposalResponses = {responses: [invalidProposalResponse]};
+	const errorProposalResponses = {errors: [queryErrorResult]};
+
+	let endorser1;
+	let endorser2;
+	let query;
 
 	beforeEach(() => {
-		failedPeers = [];
-		errorPeers = [];
-		fakeEvaluate = sinon.fake(async (peers) => {
-			const results = [];
-			for (const peer of peers) {
-				const peerName = peer.getName();
-				if (failedPeers.includes(peer)) {
-					results[peerName] = new Error(`${queryFailMessage}: ${peerName}`);
-				} else if (errorPeers.includes(peer)) {
-					results[peerName] = queryErrorResult;
-				} else {
-					results[peerName] = querySuccessResult;
-				}
-			}
-			return results;
-		});
-		stubQuery = {
-			evaluate: fakeEvaluate
-		};
-		stubPeer1 = {
-			getName: () => 'peer1'
-		};
-		stubPeer2 = {
-			getName: () => 'peer2'
-		};
-
+		query = sinon.createStubInstance(Query);
+		endorser1 = sinon.createStubInstance(Endorser);
+		endorser1.name = 'peer1';
+		endorser2 = sinon.createStubInstance(Endorser);
+		endorser2.name = 'peer2';
 	});
 
 	afterEach(() => {
 		sinon.restore();
 	});
 
 	describe('SingleQueryHandler', () => {
-		let strategy;
+		let handler;
 
 		beforeEach(() => {
-			strategy = new SingleQueryHandler([stubPeer1, stubPeer2]);
-		});
-
-		it('only queries first peer if successful', async () => {
-			await strategy.evaluate(stubQuery);
-
-			sinon.assert.calledOnce(fakeEvaluate);
-			sinon.assert.calledWith(fakeEvaluate, [stubPeer1]);
+			handler = new SingleQueryHandler([endorser1, endorser2]);
 		});
 
-		it('continues to query first peer if successful', async () => {
-			await strategy.evaluate(stubQuery);
-			await strategy.evaluate(stubQuery);
+		it('returns peer valid results', async () => {
```

Comment:
The original tests provided a spec for how each handler implementation interacted with peers while evaluating a transaction in different conditions. Specifically how each handler implementation handles failover between peers and the peers used on successive transactions. The new tests do not describe the observable behaviour of the handlers and so do not confirm that they behave correctly. The tests will have required some rework to assert the same conditions using the new underlying implementation but the scenarios tested should absolutely not be changed.